node/src/proxy_client/stream_handler_pool.rs

```diff
@@ -218,6 +243,30 @@ impl StreamHandlerPoolReal {
                     ),
                 }
             }
+            if last_data {
+                match inner.stream_writer_channels.remove(&stream_key) {
+                    Some(stream_senders) => {
+                        debug!(
+                            inner.logger,
+                            "Removing StreamWriter and Shutting down StreamReader for {:?} to {}",
+                            stream_key,
+                            stream_senders.writer_data.peer_addr()
+                        );
+                        Self::send_shutdown_signal_to_stream_reader(
+                            stream_senders.reader_shutdown_tx,
+                            &stream_key,
+                            &inner.logger,
+                        )
+                    }
+                    None => {
+                        debug!(
+                            inner.logger,
+                            "Trying to remove StreamWriter {:?}, but it's already gone", stream_key
```

Comment:
Maybe change the log to say, "Trying to remove StreamWriter and StreamReader for stream key {:?}, but it's already gone"